.github/workflows/workflow-repo-standards-validation.yml

```diff
@@ -0,0 +1,120 @@
+name: Reusable workflow to validate organization standards
+
+on:
+  workflow_call:
+
+jobs:
+  validate-name:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check repository name
+      run: |
+        repo_name="${{ github.repository }}"
+        lower_repo_name="${repo_name,,}"
+        
+        if [[ "$repo_name" != "$lower_repo_name" ]]; then
+          echo "The repository name '$repo_name' should contain only lowercase letters." >&2
+          exit 1
+        elif [[ "$repo_name" == *"_"* ]]; then
+          echo "The repository name '$repo_name' should contain dashs ( - ) and not underscores ( _ )." >&2
+          exit 1
+        fi
+      if: always()
+      shell: bash
+
+    - name: Check repository visibility
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        repo_info=$(curl -H "Authorization: token $GITHUB_TOKEN" -s "https://api.github.com/repos/${{ github.repository }}")
```

Comment:
I think you might want to consider using the official github script action https://github.com/actions/github-script that comes with a preauthenticated octokit: https://octokit.github.io/rest.js/v20/